packages/radix/src/components/Switch.tsx

```diff
@@ -1,113 +1,59 @@
+export type SwitchProps = SwitchPrimitiveProps;
 
-const marginPropNames = margin.propNames;
-
-export const Switch: FC<SwitchProps> = ({ children, ...props }) => {
-  const marginProps = pick(props, marginPropNames);
-  const inputProps = omit(props, marginPropNames);
-
-  return (
-    <SwitchWrapper {...marginProps}>
-      <Input type="checkbox" {...inputProps} />
-      <FakeSwitch />
-    </SwitchWrapper>
-  );
-};
-
-const SwitchWrapper = styled('label')<MarginProps>(
-  css({
-    position: 'relative',
-    display: 'inline-block',
-    width: 5,
-    height: 3,
-    boxSizing: 'content-box',
-    WebkitTapHighlightColor: 'rgba(0, 0, 0, 0)',
-  }),
-  margin
-);
-
-const Input = styled('input')({
-  appearance: 'none',
-  position: 'absolute',
-  top: 0,
-  left: 0,
-  width: '100%',
-  height: '100%',
-  zIndex: 0,
-  outline: 'none',
-  margin: 0,
-  opacity: 0,
-});
-
-const FakeSwitch = styled('div')(props =>
-  css({
-    width: '100%',
-    height: '100%',
-    display: 'flex',
-    alignItems: 'center',
-    position: 'relative',
-    zIndex: 1,
-    cursor: 'default',
-    userSelect: 'none',
-    '&::before': {
-      content: `''`,
-      display: 'block',
-      height: 3,
-      width: '100%',
-      backgroundColor: 'gray300',
-      borderRadius: 9999,
-      transition: 'background-color 100ms ease-out',
-    },
-    '&::after': {
-      content: `''`,
-      position: 'absolute',
-      width: '13px',
-      height: '13px',
-      top: '1px',
-      left: '1px',
-      borderRadius: 9999,
-      backgroundColor: 'white',
-      boxShadow: '0px 0px 1px rgba(0, 0, 0, 0.3), 0px 1px 2px rgba(0, 0, 0, 0.2)',
-      transition: 'background-color 100ms ease-out, transform 100ms ease-out',
-    },
-    [`${Input}:checked + &`]: {
-      '&::before': {
-        backgroundColor: 'blue600',
-      },
-      '&::after': {
-        transform: `translateX(${themeGet('space.2')(props)})`,
-      },
-    },
-    [`${Input}:not(:disabled):active + &`]: {
-      '&::before': {
-        backgroundColor: 'gray400',
-      },
-      '&::after': {
-        backgroundColor: 'gray200',
-      },
-    },
-    [`${Input}:not(:disabled):active:checked + &`]: {
-      '&::before': {
-        backgroundColor: 'blue700',
-      },
-    },
-    [`${Input}:disabled + &`]: {
-      cursor: 'not-allowed',
-      '&::before': {
-        backgroundColor: 'gray300',
-      },
-      '&::after': {
-        backgroundColor: 'gray100',
-        boxShadow: 'none',
+export const Switch = React.forwardRef<HTMLInputElement, SwitchProps>((props, forwardedRef) => (
+  <SwitchPrimitive
+    {...props}
+    ref={forwardedRef}
+    styleConfig={{
+      base: {
+        switch: {
+          normal: {
+            width: theme.sizes[5],
+            height: theme.sizes[3],
+            borderRadius: 9999,
+            backgroundColor: theme.colors.gray300,
+            transition: 'background-color 100ms ease-out',
+            'input:active:checked + &': {
```

Comment:
@vladmoroz I kept this one as a hack for now as Modulz can't do it. Happy to take it out if you think we can part ways with it.

> Note: even if we remove it, there's still a subtle change on active on the thumb.

Reply:
Yeah it’s good that it can be done like this but ive been thinking about it and personally I’d take it out. It’ll make the porting to Modulz less painful, and it’s a limitation we are all aware of

Reply:
Thanks for waiting on me @benoitgrelard, let's do the following instead of this selector and the current `:active` rule:
```
active: {
  filter: 'brightness(95%)',
},
```

However, I still think the need for the compound selector is quite high-priority product-wise. It's common to have different `:active` and / or `:focus` states on checked and not checked Switch, Checkbox, Radio. I'll track it in the Modulz repo.

Reply:
Nice solution @vladmoroz, I'll make the change.
I agree with you that compound states will still be necessary. We should probably discuss and see if/when we need to prioritise that on all levels.